pkg/templates/git.go

```diff
@@ -442,6 +442,8 @@ func syncTemplateVersion(ctx context.Context, mc model.ClientSet, tv *model.Temp
 		return err
 	}
 
+	repo.Reference = tv.Version
```

Comment:
Why do we need this? Is ref included in `tv.Source` and get parsed to `repo.Reference`?